test/parallel/test-repl-eval.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const repl = require('repl');
+
+{
+  const options = {
+    eval: common.mustCall((cmd, context) => {
+      assert.strictEqual(cmd, 'foo\n');
+      assert.deepStrictEqual(context, {animal: 'Sterrance'});
+    })
+  };
+
+  const r = repl.start(options);
+  r.context = {animal: 'Sterrance'};
+
+  try {
+    r.write('foo\n');
+  } finally {
```

Comment:
Although actually I see now that the `AssertionError`s do not cause the test to exit with an error code, so that's a problem (and apologies if that's exactly what you were referring to!).